File: frontend/src/api/dailyRewards.ts
import { api } from "@/lib/api";
import type { DailyRewardCalendar, DailyRewardClaimResult } from "@/types";

export async function fetchDailyRewardCalendar(): Promise<DailyRewardCalendar> {
  const { data } = await api.get<DailyRewardCalendar>("/daily-rewards/calendar");
  return data;
}

export async function claimDailyReward(): Promise<DailyRewardClaimResult> {
  const { data } = await api.post<DailyRewardClaimResult>("/daily-rewards/claim");
  return data;
}

File: backend/app/routers/daily_rewards.py
from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.dependencies import get_current_user
from app.database import get_db
from app.models.user import User
from app.schemas.daily_reward import DailyRewardCalendarOut, DailyRewardClaimOut
from app.services import daily_reward_service

router = APIRouter(prefix="/daily-rewards", tags=["daily-rewards"])


@router.get("/calendar", response_model=DailyRewardCalendarOut)
async def get_calendar(db: AsyncSession = Depends(get_db), user: User = Depends(get_current_user)):
    return await daily_reward_service.get_calendar(db, user)


@router.post("/claim", response_model=DailyRewardClaimOut)
async def claim_reward(db: AsyncSession = Depends(get_db), user: User = Depends(get_current_user)):
    return await daily_reward_service.claim_daily_reward(db, user)
